server/src/main/java/org/opensearch/common/cache/stats/SingleDimensionCacheStats.java

```diff
@@ -0,0 +1,283 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.cache.stats;
+
+import org.opensearch.common.metrics.CounterMetric;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * A CacheStats implementation for caches that aggregate over a single dimension, as well as holding a tier dimension.
+ * For example, caches in the IndicesRequestCache only aggregate over ShardId value.
+ */
+public class SingleDimensionCacheStats implements CacheStats {
```

Comment:
Lets not do this. In case I want to have multiple dimension, I would have to implement another class?
EhcacheDiskCache or onHeap cache should have the capability to support multiple dimensions from the start and keep it generic.

You can implement this support multiple dimensions and it can be used to have a single dimension if needed. That makes it extensible.